uui-components/src/pickers/PickerToggler.tsx

```diff
@@ -199,11 +199,16 @@ function PickerTogglerComponent<TItem, TId>(props: PickerTogglerProps<TItem, TId
         );
     };
 
+    const shouldToggleBody = (e: React.MouseEvent<HTMLDivElement>): boolean => {
+        const isInteractionDisabled = (props.isDisabled || props.isReadonly || isEventTargetInsideClickable(e));
+        const isInputFocusedAndValid = (inFocus && props.value && props.minCharsToSearch);
+        const isPickerOpenForInput = (props.isOpen && props.searchPosition === 'input');
```

Comment:
Let's better rename to 'isPickerOpenWithSearchInInput' and add comment below 'We shouldn't close the picker by click on input in toggler, because it will lead to the search clearing by body close'.

Reply:
done